### ISSUE TITLE
Views w/ domains in Enterprise are automatically being set to "isViewOverride": true

### ISSUE THREAD

**Issue by chris-fox** [Esri/solution.js#811]
When we create a view in Enterprise, we are passing the domain in the field definition. Enterprise is automatically assuming this means it is an override, even when the domain is the same as the source. This causes problems later if you attempt to modify the domain of the source service, the changes are not propagated to the view because it thinks its overridden.

In ArcGIS Online (maybe Enterprise too?) we pass "isViewOverride": true when the domain has been overridden from the source. When it is not overridden we just pass the full field definition, but don't specify "isViewOverride": false.

The fix appears to be to always pass the "isViewOverride" property and set it to true or false depending on if the field definition differs from the source. I verified that this works in ArcGIS Online as well, so we could keep the same logic in both environments.

I don't recall how we implemented this isViewOverride functionality before, but hoping it is straightforward to include passing false when it is not overridden.

As summary these are the properties that can be overridden:

alias, domain, editable, visible

**Comment by jmhauck** [comment]
@chris-fox good solution to test with?

**Comment by chris-fox** [comment]
@jmhauck, I was testing with Enterprise today using the your latest changes and I did not see the behavior of isViewOverride being specified. Road Closures is a simple solution to repro:

1. Deploy Road Closures
2. It will create a RoadClosures service and then a RoadClosures_public service, in the addToDefinition request for the _public view the fields don't contain the isViewOverride property

I noticed in Enterprise if a field contains a domain, we are passing the domain in a subsequent updateDefinition on the layer. They have changed this behavior in 11.0 and we no longer need update the domain on a view in a subsequent updateDefinition. This is true for all the override fields:

alias, domain, editable, and visible.

So instead for Enterprise we should mimic the behavior of Online for field definitions, define it in an addToDefinition and specify where it is an override or not. Do not pass any field definitions in a subsequent updateDefinition request.

**Comment by chris-fox** [comment]
@jmhauck, Just to clarify, we still need the subsequent updateDefinition for other properties of the service, just not field properties.

I also wasn't testing in the Enterprise build. I was testing using the deploySolution demo app, which should be using the latest from the develop branch. When I deployed to Online I saw the isViewOverride being passed for all fields, but when I did the same to an Enterprise portal I didn't see it being set for the fields.

https://solutionstest.geocloud.com/solution.js/demos/deploySolution/
